datalad/support/gitrepo.py

```diff
@@ -1917,6 +1907,132 @@ def _run_command_files_split(
                 err += err_
         return out, err
 
+    # Convenience wrappers for one-off git calls that don't require further
+    # processing or error handling.
+
+    def call_git(self, args, files=None,
+                 expect_stderr=False, expect_fail=False,
+                 check_fake_dates=False):
+        """Call git and return standard output.
+
+        Parameters
+        ----------
+        args : list of str
+          Arguments to pass to `git`.
+        files : list of str, optional
+          File arguments to pass to `git`. The advantage of passing these here
+          rather than as part of `args` is that the call will be split into
+          multiple calls to avoid exceeding the maximum command line length.
+        expect_stderr : bool, optional
+          Standard error is expected and should not be elevated above the DEBUG
+          level.
+        expect_fail : bool, optional
+          A non-zero exit is expected and should not be elevated above the
+          DEBUG level.
+        check_fake_dates : boolean, optional
```

Comment:
I haven't played with fake dates yet, but I wonder whether it is better (or needed) to expose this on a method-level, rather than at repo-level. In other words, would be be sufficient to put a repo instance into "fake dates modes" and have all methods act accordingly by querying an internal flag, rather than having one in each relevant method signature?

Reply:
Thanks for bringing this up.  I also don't like the idea of exposing `check_fake_dates` here.  I'll have to take a closer look at dropping it.

Reply:
> would be be sufficient to put a repo instance into "fake dates modes" and have all methods act accordingly by querying an internal flag

So this is essentially already the case.  The `.fake_dates_enabled` property involves an initial `.config.getbool` call and then the value is cached at `_fake_dates_enabled`.  I ended up just using `_git_custom_command(..., check_fake_dates=True)` for all the calls.  Here's the relevant bit from the commit message:


```
All the methods call _git_custom_command() with check_fake_dates=True.
This will be unnecessary for nearly all repositories, but not doing so
risks leaking dates in repositories configured to use fake dates.  The
other option would be to add a check_fake_dates parameter to these
methods, which is ugly because this is an obscure parameter that most
callers should not have to worry about.  Unconditionally using
check_fake_dates=True costs an attribute lookup and a
.config.getbool() call on the first use and then a _fake_dates_enabled
attribute lookup on all remaining uses for that instance.  Here are
times of an example command on the datalad repo:

  % python -m timeit -n100 \
    -s "from datalad.support.gitrepo import GitRepo" \
    -s "repo = GitRepo('.')" --  "repo.call_git(['ls-files'])"

  Results of two runs with `check_fake_dates=False`:

  1) 100 loops, best of 5: 3.87 msec per loop
  2) 100 loops, best of 5: 3.87 msec per loop

  Results of two runs with `check_fake_dates=True`:

  1) 100 loops, best of 5: 3.97 msec per loop
  2) 100 loops, best of 5: 3.96 msec per loop
```

Reply:
This looks good to me, thx.